gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,351 @@
+  private static Storage storage;
+  private static StorageOptions storageOptions;
+
+  private final Random rnd = new Random();
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    // loads the credentials from local disk as par README
+    RemoteGcsHelper gcsHelper = RemoteGcsHelper.create();
+    storageOptions = gcsHelper.options();
+    storage = storageOptions.service();
+    // create and populate test bucket
+    storage.create(BucketInfo.of(BUCKET));
+    fillFile(storage, SML_FILE, SML_SIZE);
+    fillFile(storage, BIG_FILE, BIG_SIZE);
+  }
+
+  @AfterClass
+  public static void afterClass() throws ExecutionException, InterruptedException {
+    if (storage != null && !RemoteGcsHelper.forceDelete(storage, BUCKET, 5, TimeUnit.SECONDS)) {
+      if (log.isLoggable(Level.WARNING)) {
+        log.log(Level.WARNING, "Deletion of bucket {0} timed out, bucket is not empty", BUCKET);
+      }
+    }
+  }
+
+  private static byte[] randomContents(int size) {
+    byte[] bytes = new byte[size];
+    new Random(size).nextBytes(bytes);
+    return bytes;
+  }
+
+  private static void fillFile(Storage storage, String fname, int size) throws IOException {
+    storage.create(BlobInfo.builder(BUCKET, fname).build(), randomContents(size));
+  }
+
+  @Test
+  public void testFileExists() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.exists(path)).isTrue();
+  }
+
+  @Test
+  public void testFileSize() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.size(path)).isEqualTo(SML_SIZE);
+  }
+
+  @Test(timeout=60_000)
+  public void testReadByteChannel() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    long size = Files.size(path);
+    SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+    assertThat(chan.size()).isEqualTo(size);
+    ByteBuffer buf = ByteBuffer.allocate(SML_SIZE);
+    int read = 0;
+    while (chan.isOpen()) {
+      int rc = chan.read(buf);
+      assertThat(chan.size()).isEqualTo(size);
+      if (rc < 0) {
+        // EOF
+        break;
+      }
+      assertThat(rc).isGreaterThan(0);
+      read += rc;
+      assertThat(chan.position()).isEqualTo(read);
+    }
+    assertThat(read).isEqualTo(size);
+    byte[] expected = new byte[SML_SIZE];
+    new Random(SML_SIZE).nextBytes(expected);
+    assertThat(Arrays.equals(buf.array(), expected)).isTrue();
+  }
+
+  @Test
+  public void testSeek() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(BIG_FILE);
+    int size = BIG_SIZE;
+    byte[] contents = randomContents(size);
+    byte[] sample = new byte[100];
+    byte[] wanted, wanted2;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://www.codacy.com/app/mziccard/gcloud-java/file/2295976424/issues/source?bid=3123107&fileBranchId=3162887#l159)